`tests/test_validators.py::test_assert_raises_validation_error` fails with pytest 8

### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

The following test failures with pytest 8 remained:

```
FAILED tests/test_validators.py::test_assert_raises_validation_error - assert [{'type': 'assertion_error', 'loc': ('a',), 'msg': "Assertion failed, invalid a\nassert 'snap' == 'a'\n  \n  - a\n  + snap"...
```

### Example Code

```Python
$ python -m pytest -Wdefault -vv
============================= test session starts ==============================
platform linux -- Python 3.11.8, pytest-8.1.1, pluggy-1.4.0 -- /tmp/pydantic/.venv/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase(PosixPath('/tmp/pydantic/.hypothesis/examples'))
benchmark: 4.0.0 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=True warmup_iterations=100000)
rootdir: /tmp/pydantic
configfile: pyproject.toml
testpaths: tests
plugins: hypothesis-6.99.5, xdist-3.5.0, benchmark-4.0.0, examples-0.0.10, mock-3.12.0, Faker-24.1.0
collecting ... collected 5002 items

[…]
============================================================== FAILURES ===============================================================
_________________________________________________ test_assert_raises_validation_error _________________________________________________

    def test_assert_raises_validation_error():
        class Model(BaseModel):
            a: str
    
            @field_validator('a')
            @classmethod
            def check_a(cls, v: Any):
                assert v == 'a', 'invalid a'
                return v
    
        Model(a='a')
    
        with pytest.raises(ValidationError) as exc_info:
            Model(a='snap')
        injected_by_pytest = "assert 'snap' == 'a'\n  - a\n  + snap"
>       assert exc_info.value.errors(include_url=False) == [
            {
                'ctx': {'error': HasRepr(repr(AssertionError("invalid a\nassert 'snap' == 'a'\n  - a\n  + snap")))},
                'input': 'snap',
                'loc': ('a',),
                'msg': f'Assertion failed, invalid a\n{injected_by_pytest}',
                'type': 'assertion_error',
            }
        ]
E       assert [{'type': 'assertion_error', 'loc': ('a',), 'msg': "Assertion failed, invalid a\nassert 'snap' == 'a'\n  \n  - a\n  + snap", 'input': 'snap', 'ctx': {'error': AssertionError("invalid a\nassert 'snap' == 'a'\n  \n  - a\n  + snap")}}] == [{'ctx': {'error': HasRepr('AssertionError("invalid a\\nassert \'snap\' == \'a\'\\n  - a\\n  + snap")')}, 'input': 'snap', 'loc': ('a',), 'msg': "Assertion failed, invalid a\nassert 'snap' == 'a'\n  - a\n  + snap", 'type': 'assertion_error'}]
E         
E         At index 0 diff: {'type': 'assertion_error', 'loc': ('a',), 'msg': "Assertion failed, invalid a\nassert 'snap' == 'a'\n  \n  - a\n  + snap", 'input': 'snap', 'ctx': {'error': AssertionError("invalid a\nassert 'snap' == 'a'\n  \n  - a\n  + snap")}} != {'ctx': {'error': HasRepr('AssertionError("invalid a\\nassert \'snap\' == \'a\'\\n  - a\\n  + snap")')}, 'input': 'snap', 'loc': ('a',), 'msg': "Assertion failed, invalid a\nassert 'snap' == 'a'\n  - a\n  + snap", 'type': 'assertion_error'}
E         
E         Full diff:
E           [
E               {
E                   'ctx': {
E         -             'error': HasRepr('AssertionError("invalid a\\nassert \'snap\' == \'a\'\\n  - a\\n  + snap")'),
E         ?                      ---------                         -         -     -     -  -          -           --
E         +             'error': AssertionError("invalid a\nassert 'snap' == 'a'\n  \n  - a\n  + snap"),
E         ?                                                                      +++
E                   },
E                   'input': 'snap',
E                   'loc': (
E                       'a',
E                   ),
E                   'msg': 'Assertion failed, invalid a\n'
E                   "assert 'snap' == 'a'\n"
E         +         '  \n'
E                   '  - a\n'
E                   '  + snap',
E                   'type': 'assertion_error',
E               },
E           ]

tests/test_validators.py:1307: AssertionError
```


### Python, Pydantic & OS Version

```Text
pydantic version: 2.7.0a1
        pydantic-core version: 2.16.3
          pydantic-core build: profile=release pgo=true
                 install path: /tmp/pydantic/pydantic
               python version: 3.11.8 (main, Feb 11 2024, 17:30:38) [GCC 13.2.1 20240113]
                     platform: Linux-6.7.9-gentoo-dist-x86_64-AMD_Ryzen_5_3600_6-Core_Processor-with-glibc2.39
             related packages: typing_extensions-4.10.0
                       commit: 18d39feb
```

--- a/tests/test_validators.py
+++ b/tests/test_validators.py
@@ -1344,20 +1344,20 @@ def test_assert_raises_validation_error():
         @field_validator('a')
         @classmethod
         def check_a(cls, v: Any):
-            assert v == 'a', 'invalid a'
+            if v != 'a':
+                raise AssertionError('invalid a')
             return v
 
     Model(a='a')
 
     with pytest.raises(ValidationError) as exc_info:
         Model(a='snap')
-    injected_by_pytest = "assert 'snap' == 'a'\n  - a\n  + snap"
     assert exc_info.value.errors(include_url=False) == [
         {
-            'ctx': {'error': HasRepr(repr(AssertionError("invalid a\nassert 'snap' == 'a'\n  - a\n  + snap")))},
+            'ctx': {'error': HasRepr(repr(AssertionError('invalid a')))},
             'input': 'snap',
             'loc': ('a',),
-            'msg': f'Assertion failed, invalid a\n{injected_by_pytest}',
+            'msg': 'Assertion failed, invalid a',
             'type': 'assertion_error',
         }
     ]

